test/helpers/kubectl.go

```diff
@@ -105,39 +108,59 @@ func GetCurrentIntegration() string {
 // commands on the node which is accessible via the SSH metadata stored in its
 // SSHMeta.
 type Kubectl struct {
-	*SSHMeta
+	Executor
 	*serviceCache
+	BasePath string
 }
 
 // CreateKubectl initializes a Kubectl helper with the provided vmName and log
 // It marks the test as Fail if cannot get the ssh meta information or cannot
 // execute a `ls` on the virtual machine.
 func CreateKubectl(vmName string, log *logrus.Entry) *Kubectl {
-	node := GetVagrantSSHMeta(vmName)
-	if node == nil {
-		ginkgoext.Fail(fmt.Sprintf("Cannot connect to vmName  '%s'", vmName), 1)
-		return nil
+	if config.CiliumTestConfig.Kubeconfig == "" {
```

Comment:
Since `config.CiliumTestConfig.Kubeconfig` is the only global here, maybe just pass it in? When we come to change this (again) we'll have 1 less thing to think about.

Reply:
But by doing this we are leaking, no? The import tree now includes the config package. It's not that important, though.